03week/towersOfHanoi.js

```diff
@@ -19,27 +35,99 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+const movePiece = (startStack,endStack) => {
+  // made  a copy of stacks so i wouldnt mutate the original stacks
+  const stacksCopy = {        
+    a:[],
+    b:[],
+    c:[]
+  
+  }
+  // getting keys from stacks, and copying the value of each key into the copy
+  //using bracket notation to fetch key in string format
+  Object.keys(stacks).forEach( key => {
+    stacksCopy[key] = stacks[key]
+  })
+  // the array with the starting numbers
+  const startingStack = stacksCopy[startStack]
+  // where the block ends up
+  const endingStack = stacksCopy[endStack]
+  // get the value of the last block , so i can move it to the ending stack
+  const currentBlock = startingStack.pop()
+  // pushing it to the ending block
+  endingStack.push(currentBlock);
+  // setting it to the variable the game cares about
+  stacks = stacksCopy
+  
 }
+const  isValid =(startStack)=>{
+  // if input is a,b,or c
+
+  const emptyStartStack = stacks[startStack][stacks[startStack].length === 0];
+  //is the start stack empty? (if it is , === invalid)
+  if (emptyStartStack) {
+    return false
+  }else {
+    return true
+  }
+  // if(stacks[startStack].length == 0){
+  //   console.log('You chose an empty start stack, please choose a start stack with numbers.');
+  //   return false;
+  // } else if(stacks[endStack].length == 0){
+  //   return true;
+  // }
 
-function isLegal() {
-  // Your code here
 
+}
+const isLegalMove = (startStack,endStack) => {
+  // // i cant get this to work, and i dont know whyyyyyyyy
+  // just kidding, i had to make varibles to hold the length and get last index values
+  
+
+// //is block being moved smaller than last block on finish stack
+  const greater = stacks[startStack][stacks[startStack].length - 1];
+  const lesser = stacks[endStack][stacks[endStack].length - 1];
+  if (lesser < greater) {
+    return false;
+  } else {
+    return true;
+    
+  }
+   
+  
 }
 
-function checkForWin() {
-  // Your code here
-
+const checkForWin = () => {
+  // checks to see if stacks[c] has all of the 'blocks', if so the game is won
+  if (stacks['c'].length === 4 ){
+    return true}   
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
+const resetGame = () => {
+  // calling stack var to reset the game to its original state if game is won
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
+}
 
+const towersOfHanoi = (startStack, endStack) => {
+  // calling all my functions to get this game workin, 
+  // except not all of my functions werk
+  
+  if (isLegalMove(startStack,endStack)){
+    movePiece(startStack,endStack);}
+  else if(!isLegalMove(startStack, endStack)){
+    console.log('You can only put a smaller number on a larger number.')
+  }
+  if (checkForWin()) {
```

Comment:
you're checking for a win even if the move isn't valid, change this function organization